### ISSUE TITLE
React Components rendered as string

### ISSUE THREAD

**Issue by besrabasant** [cerebral/marksy#20]
I have Component that I am using with markdown.

When I have only the component in the markdown it is rendered correctly.
```
 const compiled = compile('<VariableInput  />');
```

But when I have some more text in the markdown. The component is rendered as plain text
```
 const compiled = compile('Hello World <VariableInput />');
```

rendered as 

```
Hello World <VariableInput />
```

Please help me through this

**Comment by christianalfoni** [comment]
@besrabasant It seems that `marked` (markdown parser under the hood) is evaluating HTML on same line as text... as text. So if you do:

```js
 const compiled = compile('Hello World\n<VariableInput />');
```

It will probably work. I would suggest using marksy language though:

```js
 const compiled = compile('Hello World ```marksy\n<VariableInput />```');
```

I guess the input would come from a textarea or similar, so typically you do not have to worry about these linebreaks, but for the purpose of your example here it is needed :-)

**Comment by besrabasant** [comment]
@christianalfoni actually I want the Input to be rendered inline, that is on the same line as the text and the input is from `<div contentEditable={true}></div>`....

**Comment by christianalfoni** [comment]
@besrabasant Aha, I do not think that is possible, because `marked` renders inline text as text. It can not inline html with paragraph. 

From the docs:
**Inline level renderer methods**
strong(string text)
em(string text)
codespan(string code)
br()
del(string text)
link(string href, string title, string text)
image(string href, string title, string text)

HTML is not part of that list

**Comment by besrabasant** [comment]
@christianalfoni I need to ask one more thing will this work with [react-in-markdown](https://github.com/kitze/react-in-markdown) as the example uses [markdown-to-react-components](https://github.com/christianalfoni/markdown-to-react-components) and you say that  repo is deprecated and instead use this repo.

**Comment by christianalfoni** [comment]
Ah, `react-in-markdown` is indeed based off of the old `markdown-to-react-components`. @kitze was planning to coop with me on `marksy` to bring his ideas from `react-in-markdown` to this new `marksy` project, but he was in lack of time.

But yeah, generally writing HTML inline does not work, because of the underlying `marked` library used in all of these projects. I would suggest it is a markdown general feature, cause inlining HTML with text does not seem like what you would normally want :)

**Comment by christianalfoni** [comment]
BUT! You can rather create a component to handle it:

```js
<TextWithComponents>Hello World <VariableInput /></TextWithComponents>
```

:)

**Comment by christianalfoni** [comment]
Just a component that returns its children

**Comment by Guria** [comment]
can't it be just `<p>Hello World <VariableInput /></p>` then?

**Comment by besrabasant** [comment]
@Guria Is thought so but it isn't working

**Comment by christianalfoni** [comment]
Hm, yeah, that should also work... 

But @besrabasant , you are importing `import marksy from 'marksy/components'`? Not just `marksy`?

**Comment by besrabasant** [comment]
yes I am importing `import marksy from 'marksy/components'`

**Comment by besrabasant** [comment]
@christianalfoni I tried `<TextWithComponents>Hello World <VariableInput /></TextWithComponents>`

but its not working

Here's my setup
```
import React from "react";
import marksy from 'marksy/components';
import VariableInput from "../VariableInput";

 const compile = marksy({
        createElement,
        components: {
            VariableInput,
            MyText (props) {
                return <div>{props.children}</div>
            }
        }
    });
    const compiled = compile('<MyText> <VariableInput /> \n[Google](www.google.com)</MyText>',null);

console.log(compiled.tree);
```
The `tree` array contains just one element of type `a`. Am I doing it correct.

**Comment by besrabasant** [comment]
@christianalfoni @Guria Finally got inline components working using [react-components-in-markdown](https://www.npmjs.com/package/react-components-in-markdown) and [markdown-to-react-components](https://github.com/christianalfoni/markdown-to-react-components).... but with a warning 
```
Warning: Accessing createClass via the main React package is deprecated, and will be removed in React v16.0. Use a plain JavaScript class instead. If you're not yet ready to migrate, create-react-class v15.* is available on npm as a temporary, drop-in replacement. For more info see https://fb.me/react-create-class
```
How can I fix this???

**Comment by christianalfoni** [comment]
@besrabasant Then there is a bug in `marksy` we should fix :-) Looking into this now

**Comment by christianalfoni** [comment]
Okay, it is fixed, 3.1.2 :-)
